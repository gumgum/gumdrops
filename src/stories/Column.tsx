import React, { ReactChild, ReactChildren } from 'react';

type ColumnValue = string | number;

export interface ColumnProps extends React.HTMLAttributes<Element> {
    xs?: ColumnValue;
    sm?: ColumnValue;
    md?: ColumnValue;
    lg?: ColumnValue;
    xl?: ColumnValue;
}

export const Column: React.FC<ColumnProps> = ({
    xs,
    sm,
    md,
    lg,
    xl,
    className,
    children,
    ...otherProps
}) => {
    const sizes = { xs, sm, md, lg, xl };

    const classList = Object.keys(sizes)
        .reduce((list, breakpoint) => {
            const size = sizes[breakpoint];
            return list.concat(size ? `gds-layout__column--${breakpoint}-${size}` : []);
        }, [])
        .concat(className || [])
        .join(' ');

    return (
        <div className={classList} {...otherProps}>
            {children}
        </div>
    );
};
