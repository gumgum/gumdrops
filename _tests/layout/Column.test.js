import renderer from 'react-test-renderer';
import React from 'react';
import Column from '../../components/layout/Column';

const props = {
    xs: 12,
    sm: 6,
    md: 4,
    lg: 3,
    xl: 2
};

test('Expect <Column> to render properly', () => {
    const tree = renderer
        .create(
            <Column {...props}>
                <div>Content</div>
            </Column>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
