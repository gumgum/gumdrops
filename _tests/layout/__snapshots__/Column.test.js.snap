// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Column> to render properly 1`] = `
<div
  className="gds-layout__column--xs-12 gds-layout__column--sm-6 gds-layout__column--md-4 gds-layout__column--lg-3 gds-layout__column--xl-2"
>
  <div>
    Content
  </div>
</div>
`;
